Ghidra/Framework/Generic/src/main/java/resources/ResourceManager.java

```diff
@@ -466,11 +466,43 @@ public static ImageIcon loadImage(String imageName, byte[] imageBytes) {
 	 * @return the scaled image.
 	 */
 	public static ImageIcon loadImage(String filename, int width, int height) {
-		ImageIcon loadImage = loadImage(filename);
+
+		ImageIcon icon = iconMap.get(filename);
+		if (icon != null) {
+			return icon;
+		}
+		File imageFile = new File(filename);
+		if (imageFile.exists()) {
+			try {
+				icon = new ScaledImageIconWrapper(imageFile.toURI().toURL(), width, height);
+				iconMap.put(filename, icon);
+				return icon;
+			}
+			catch (MalformedURLException e) {
+				// handled below
+			}
+		}
+		URL url = getResource(filename);
+		if (url != null) {
+			icon = new ScaledImageIconWrapper(url, width, height);
+			iconMap.put(filename, icon);
+			return icon;
+		}
+		return getDefaultIcon();
+	}
+
+	/**
+	 * Load and scale the image specified by filename; returns null if problems occur trying to load
+	 * the file. The height and width of the picture are fixed, 16*16.
+	 * @param filename name of file to load, e.g., "images/home.gif"
+	 * @return the scaled image.
+	 */
+	public static ImageIcon loadImage(String filename) {
+		ImageIcon loadImage = loadImage(filename, 16, 16);
```

Comment:
I don't think hardcoding the size in the internal implementation is a good idea.. This might cause unintended problem where some interface intended to have the whole image loaded but only has it loaded as 16 * 16. A simple solution would be adding a new interface called something like "loadImageIconSized" or just "loadIconDefaultSize" and let those that wanted the default size to be used in this way instead forcing everything here.